sklearn/decomposition/nmf.py

```diff
@@ -277,30 +632,208 @@ class ProjectedGradientNMF(BaseEstimator, TransformerMixin):
             'nndsvdar': NNDSVD with zeros filled with small random values
                 (generally faster, less accurate alternative to NNDSVDa
                 for when sparsity is not desired)
-            'random': non-negative random matrices
+            'custom': use custom matrices W and H
+
+    update_H : boolean, default: True
+        Set to True, both W and H will be estimated from initial guesses.
+        Set to False, only W will be estimated.
+
+    solver : 'proj-grad' | 'coordinate'
+        Numerical solver to use.
+
+    tol : float, default: 1e-4
+        Tolerance of the stopping condition.
+
+    max_iter : integer, default: 200
+        Maximum number of iterations before timing out.
+
+    alpha : double, default: 0.
+        Constant that multiplies the regularization terms.
+
+    l1_ratio : double, default: 0.
+        The regularization mixing parameter, with 0 <= l1_ratio <= 1.
+        For l1_ratio = 0 the penalty is an L2 penalty.
+        For l1_ratio = 1 it is an L1 penalty.
+        For 0 < l1_ratio < 1, the penalty is a combination of L1 and L2.
+
+    regularization : 'both' | 'components' | 'transformation' | None
+        Select whether the regularization affects the components (H), the
+        transformation (W), both or none of them.
+
+    random_state : integer seed, RandomState instance, or None (default)
+        Random number generator seed control.
+
+    verbose : integer, default: 0
+        The verbosity level.
+
+    shuffle : boolean
+        If True, the samples will be taken in shuffled order during
+        coordinate descent.
+
+    nls_max_iter : integer, default: 2000
+        Number of iterations in NLS subproblem.
+        Used only in deprecated proj-grad solver.
 
     sparseness : 'data' | 'components' | None, default: None
         Where to enforce sparsity in the model.
+        Used only in deprecated proj-grad solver.
```

Comment:
s/in/in the/

also `proj-grad` should match the actual name the solver ends up having.